src/VisualStudio/Core/Def/Implementation/TableDataSource/AbstractTableEntriesSnapshot.cs

```diff
@@ -169,14 +175,32 @@ protected static bool TryNavigateTo(Workspace workspace, DocumentId documentId,
         protected bool TryNavigateToItem(int index, bool previewTab, bool activate, CancellationToken cancellationToken)
         {
             var item = GetItem(index);
-            if (item is not { DocumentId: { } documentId })
-            {
+            if (item is null)
                 return false;
-            }
 
             var workspace = item.Workspace;
             var solution = workspace.CurrentSolution;
-            var document = solution.GetDocument(documentId);
+            if (item is not { DocumentId: { } documentId })
+            {
+                if (item is { ProjectId: { } projectId }
+                    && solution.GetProject(projectId) is { } project)
+                {
+                    // We couldn't find a document ID when the item was created, so it may be a source generator
+                    // output.
+                    var documents = ThreadingContext.JoinableTaskFactory.Run(() => project.GetSourceGeneratedDocumentsAsync(cancellationToken).AsTask());
+                    var projectDirectory = Path.GetDirectoryName(project.FilePath);
```

Comment:
❓ I didn't understand this question

Reply:
What produced the string being returned in item.GetOriginalFilePath?

Reply:
I believe it could be any of the following:

* A build error in the Build view of error list
* A build error translated to the Build+IntelliSense view of error list
* An IntelliSense error in the Build+IntelliSense or IntelliSense views of the error list

Reply:
It seems to always give a fake path. My local testing that derived this approach was performed in a project that customizes the output paths (to a location different than what appears in the error list).

Reply:
I am wondering if we can move all this work to identify the source generated document ID from the reported document file path to the place where we create the DiagnosticData. For example, for build errors, this should likely go into ProjectExternalErrorReporter.